### PR TITLE
automate-remote-backend

### PR DESCRIPTION
**Pull request by MoYVR**

### CODE REVIEW

File: level0/main.tf [hunk 1]
```diff
@@ -0,0 +1,28 @@
+
+resource "aws_s3_bucket" "main" {
```
Comment (vladimirmukhin) [inline level0/main.tf]:
```suggestion
resource "aws_s3_bucket" "main" {
```

File: level0/main.tf [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  bucket = "terraformstatelabmo"
+
+  tags = {
+    Name = "main"
+  }
+}
+
+
+resource "aws_s3_bucket" "terraform_remote_state" {
```
Comment (vladimirmukhin) [inline level0/main.tf]:
```suggestion
}

resource "aws_s3_bucket" "terraform_remote_state" {
```

File: level0/main.tf [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  write_capacity = 1
+  hash_key       = "LockID"
+
+  attribute {
+    name = "LockID"
+    type = "S"
+  }
+
+}
+
```
Comment (vladimirmukhin) [inline level0/main.tf]:
```suggestion
  }
}
```

File: level0/provider.tf [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.16"
+    }
+  }
+
+  required_version = ">= 1.2.0"
+
+}
```
Comment (vladimirmukhin) [inline level0/provider.tf]:
```suggestion
  required_version = ">= 1.2.0"
}
```

File: level0/provider.tf [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+# Region
+
+provider "aws" {
```
Comment (vladimirmukhin) [inline level0/provider.tf]:
```suggestion
provider "aws" {
```

File: level0/terraform.tfstate [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+  "version": 4,
+  "terraform_version": "1.2.7",
+  "serial": 8,
+  "lineage": "6119ce0e-a54a-afe1-2811-2e6bec611731",
+  "outputs": {},
+  "resources": []
+}
+
```
Comment (vladimirmukhin) [inline level0/terraform.tfstate]: This state does not contain anything. Pllease run terraform apply in level0 and after that commit the generated state file.

Also, you don't need to run terraform destroy. It is ok to keep resources from level0 - they will not cost anything

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

File: level0/terraform.tfstate (hunk 6 above)
Comment (MoYVR) [reply level0/terraform.tfstate]: thanks for the tip!

Comment (MoYVR) [reply level0/terraform.tfstate]: the file got updated

**Review by vladimirmukhin — APPROVED** [review]

**Pull request merged**